File: 2nd year/MPP/project-mpp/java-project/TriathlonModel/src/main/java/triathlon/model/Result.java
package triathlon.model;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import javax.persistence.Entity;
import java.io.Serializable;

@Entity
@Table(name = "Result")
public class Result extends triathlon.model.Entity<Long> implements Serializable {
    private Game game;
    private Athlete athlete;
    private Float value;

    public Result(Long id, Game game, Athlete athlete, Float value) {
        setId(id);
        this.game = game;
        this.athlete = athlete;
        this.value = value;
    }

    public Result(Game game, Athlete athlete, Float value) {
        this.game = game;
        this.athlete = athlete;
        this.value = value;
    }

    public Result() {}

    @Override
    public void setId(Long id) {
        super.setId(id);
    }

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @GenericGenerator(name="identity", strategy = "identity")
    @Override
    public Long getId() {
        return super.getId();
    }

    @ManyToOne
    @JoinColumn(name = "gameId", referencedColumnName = "id")
    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    @ManyToOne
    @JoinColumn(name = "athleteId", referencedColumnName = "id")
    public Athlete getAthlete() {
        return athlete;
    }

    public void setAthlete(Athlete athlete) {
        this.athlete = athlete;
    }

    @Column(name = "value")
    public Float getValue() {
        return value;
    }

    public void setValue(Float value) {
        this.value = value;
    }
}

File: 2nd year/MPP/project-mpp/csharp-project/model/Result.cs
namespace model
{
    public class Result : Entity<long>
    {
        public Game Game { get; set; }
        public Athlete Athlete { get; set; }
        public double Value { get; set; }

       /* public Result(Game game, Athlete athlete, double value) : this(0L, game, athlete, value)
        {

        }

        public Result(long id, Game game, Athlete athlete, double value)
        {
            Id = id;
            Game = game;
            Athlete = athlete;
            Value = value;
        }
       */
        public override string ToString()
        {
            return Athlete.Name + " " + Value;
        }
    }
}
